src/components/TextField/TextField.scss

```diff
@@ -193,6 +200,19 @@ $stacking-order: (
   padding-bottom: spacing(tight);
 }
 
+.ClearButton {
+  z-index: z-index(contents, $stacking-order);
+  appearance: none;
+  border: none;
+  padding: 0;
+  margin: 0 spacing(tight);
+  cursor: pointer;
+
+  svg:hover {
+    fill: color('ink', 'lighter');
```

Comment:
Should the hover be on the SVG itself? And what about the focus state (if applicable)?

Also, should this SVG be colored using `@include color-icon('ink', 'lighter');`? 

See https://github.com/Shopify/polaris-react/blob/570d4293256f1da47e5bf311115dc1e89b7454ec/src/components/Icon/Icon.scss#L6-L14